pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -468,6 +468,13 @@ void downloadAndReplaceSegment(String segmentName, SegmentZKMetadata segmentZKMe
 
   private void downloadSegmentFromDeepStore(String segmentName, IndexLoadingConfig indexLoadingConfig, String uri) {
     File segmentTarFile = new File(_indexDir, segmentName + TarGzCompressionUtils.TAR_GZ_FILE_EXTENSION);
```

Comment:
I see a similar method in BaseTableManager, which does this to prepare a temp root dir for downloading and uncompressing etc..
```
File tempRootDir = getTmpSegmentDataDir("tmp-" + segmentName + "-" + UUID.randomUUID());
```

In RealtimeTableDataManager, it actually also uses this util `getTmpSegmentDataDir` but inside `untarAndMoveSegment()` a few lines below. So maybe, we make those two places a bit more consistent by creating a tempRootDir.

But to make things robust, inside getTmpSegmentDataDir(), we do the check you added here to recreate tmpRootDir if it used to exist.

Reply:
this code is very difficult to unit test, so for the sake of not creating any other side effects, I would prefer to leave it at this simple check.. I don't feel confident moving the segmentTarFile into a tmp dir and not breaking something else..